TestDocsServesHTML fails locally when Hugo output isn't pre-built

## Observed

\`\`\`
--- FAIL: TestDocsServesHTML (0.24s)
    docs_test.go:50: served the docs stub placeholder — run \`hugo --minify --baseURL /docs/\` from docs/ and copy public/ to internal/docs/dist/ before testing (CI unit job does this automatically)
\`\`\`

Reproduces on clean \`main\` (b6b70f5) with \`go test ./internal/api\` — unrelated to any in-flight work. CI passes because the unit job runs Hugo first (\`.github/workflows/ci.yml\` installs peaceiris/actions-hugo\@v3 and builds \`docs/public\`); local and sandbox test runs without that prep trip the stub detector in \`internal/api/docs_test.go:50\`.

## Why this matters

Every developer running the full test suite locally hits this, noise during triage (real failures get lost next to a reliable FAIL). \`/debug\` and \`go test ./...\` can't be trusted without out-of-band knowledge that you must run Hugo first.

## Options (one should be picked, not all)

1. Have the test detect the stub output and \`t.Skip\` with the same diagnostic message instead of \`t.Fatal\`. Local runs stay green; CI still catches the regression because it \*does\* build docs first, so the stub condition only fires when the pipeline skipped Hugo — which would already fail other docs jobs.
2. Make the test shell out to \`hugo\` on-demand and build into a tempdir. Cost: every \`internal/api\` test run invokes Hugo (~1s) and requires \`hugo\` + the \`docs/themes/hugo-book\` submodule locally. Catches real regressions but penalizes every unit run.
3. Document the \`hugo ...\` prep as a prerequisite in AGENTS.md / README and leave the test as-is. Cheapest but doesn't stop the next person from hitting it.

Leaning toward (1): CI already guarantees the docs build runs before the test, so the stub-detector condition is redundant defense in local runs. Happy to implement whichever you pick.

## Hypothesis

The stub-vs-real-output check was added as a fail-fast guard against shipping a broken docs bundle (commit history for \`docs_test.go\` will say why). It doesn't need to fire when the local dev path legitimately has no Hugo output — that's a setup gap, not a regression worth blocking on.

--- a/internal/api/docs_test.go
+++ b/internal/api/docs_test.go
@@ -4,6 +4,7 @@ import (
 	"context"
 	"net/http"
 	"net/http/httptest"
+	"os"
 	"strings"
 	"testing"
 )
@@ -41,12 +42,18 @@ func TestDocsServesHTML(t *testing.T) {
 	}
 
 	body := rec.Body.String()
-	// The stub at internal/docs/dist/index.html ships when Hugo
-	// hasn't run. Skip locally so `go test ./...` stays green without
-	// out-of-band setup; CI's unit job builds docs first, so a real
-	// regression still fires the bk-hero check below.
+	// Stub at internal/docs/dist/index.html indicates the Hugo build
+	// step was skipped. In CI this is a contract violation — the unit
+	// job's Hugo prep must run so TestDocsServesHTML exercises the
+	// real embed (the gap that let broken docs reach prod before).
+	// Locally, skip so `go test ./...` stays green without requiring
+	// every developer to run Hugo first.
 	if strings.Contains(body, "Documentation was not included in this build") {
-		t.Skip("docs stub served — run `hugo --minify --baseURL /docs/` from docs/ and copy public/ to internal/docs/dist/ to exercise this test (CI unit job does this automatically)")
+		const msg = "docs stub served — run `hugo --minify --baseURL /docs/` from docs/ and copy public/ to internal/docs/dist/ to exercise this test (CI unit job does this automatically)"
+		if os.Getenv("CI") != "" {
+			t.Fatal(msg)
+		}
+		t.Skip(msg)
 	}
 	// Marker from docs/layouts/index.html; only present when Hugo
 	// produced real output with the hugo-book theme submodule. Also

--- a/internal/api/docs_test.go
+++ b/internal/api/docs_test.go
@@ -41,13 +41,12 @@ func TestDocsServesHTML(t *testing.T) {
 	}
 
 	body := rec.Body.String()
-	// Guard against the "docs not included" stub at
-	// internal/docs/dist/index.html being served because the Hugo
-	// build step was skipped. The stub is valid HTML, so status +
-	// content-type alone passes silently — that gap is exactly how
-	// broken docs reached production before.
+	// The stub at internal/docs/dist/index.html ships when Hugo
+	// hasn't run. Skip locally so `go test ./...` stays green without
+	// out-of-band setup; CI's unit job builds docs first, so a real
+	// regression still fires the bk-hero check below.
 	if strings.Contains(body, "Documentation was not included in this build") {
-		t.Fatal("served the docs stub placeholder — run `hugo --minify --baseURL /docs/` from docs/ and copy public/ to internal/docs/dist/ before testing (CI unit job does this automatically)")
+		t.Skip("docs stub served — run `hugo --minify --baseURL /docs/` from docs/ and copy public/ to internal/docs/dist/ to exercise this test (CI unit job does this automatically)")
 	}
 	// Marker from docs/layouts/index.html; only present when Hugo
 	// produced real output with the hugo-book theme submodule. Also

